test/e2e_yaml_test.go

```diff
@@ -0,0 +1,238 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"io"
+	"io/ioutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	knativetest "knative.dev/pkg/test"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+const (
+	DEFAULT_KO_DOCKER_REPO = `gcr.io\/christiewilson-catfactory`
+	DEFAULT_NAMESPACE      = `namespace: default`
+)
+
+// Generalized for tekton CRDs
+func GetCreatedTektonCrd(input []byte, kind string) (string, error) {
+	re := regexp.MustCompile(kind + `.tekton.dev\/(.+) created`)
+	submatch := re.FindSubmatch(input)
+	if submatch == nil {
+		return "", nil
+	}
+	return string(submatch[1]), nil
+}
+
+func ValidatePipelineRunDone(t *testing.T, c *clients, pipelineRunName string) {
+	err := WaitForPipelineRunState(c, pipelineRunName, pipelineRunTimeout, Succeed(pipelineRunName), pipelineRunName)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	return
+}
+
+func ValidateTaskRunDone(t *testing.T, c *clients, taskRunName string) {
+	// Per test basis
+	err := WaitForTaskRunState(c, taskRunName, Succeed(taskRunName), taskRunName)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	return
+}
+
+// Substitute docker repos/bucket paths from environment to allow tests on
+// local clusters. Also update namespaces so that examples with ServiceAccounts
+// work under test
+func SubEnv(input []byte, namespace string) ([]byte, error) {
+	val, ok := os.LookupEnv("KO_DOCKER_REPO")
+	var output []byte
+	if ok {
+		re := regexp.MustCompile(DEFAULT_KO_DOCKER_REPO)
+		output = re.ReplaceAll(input, []byte(val))
+	} else {
+		return input, nil
+	}
+
+	re := regexp.MustCompile(DEFAULT_NAMESPACE)
+	output = re.ReplaceAll(output, []byte(strings.ReplaceAll(DEFAULT_NAMESPACE, "default", namespace)))
+	return output, nil
+}
+
+func KoCreate(input []byte, namespace string) ([]byte, error) {
+	cmd := exec.Command("ko", "create", "-n", namespace, "-f", "-")
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return []byte("Cannot create Stdin Pipe"), err
+	}
+
+	go func() {
+		defer stdin.Close()
+		io.WriteString(stdin, string(input))
+	}()
+
+	out, err := cmd.CombinedOutput()
+	return out, err
+}
+
+func DeleteClusterTask(t *testing.T, c *clients, name string) {
+	t.Logf("Deleting clustertask %s", name)
+	err := c.ClusterTaskClient.Delete(name, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func YamlSubtest(path string, validateFunc func(t *testing.T, c *clients, name string), kind string) func(subt *testing.T) {
+	return func(subt *testing.T) {
+		subt.Parallel()
+
+		// Setup unique namespaces for each test so they can run in complete
+		// isolation
+		c, namespace := setup(subt)
+
+		adminAvailable := true
+
+		// Same check as in github.com/tektoncd/plumbing/scripts/e2e-tests.sh
```

Comment:
im a bit confused about this whole admin check, i didnt see that as something required for the yaml tests when i worked on the other PR 🤔 

a couple requests:
- do we absolutely need this?
- if we do, could we keep this in a separate function with some docstrings/comments explaining whats going on and why?

Reply:
- Do we need this?
   - I believe the `pipelinerun.yaml` example tests (both alpha and beta) requested cluster-admin permissions in the comments as they were adding `ClusterRoleBinding`s. In conjunction with a namespace re-write (see [here](https://github.com/tektoncd/pipeline/pull/2685#discussion_r434211607)), it fixed the two tests for me. I will confirm to be sure, perhaps only the namespace rewrite is required.

If we do need this, I will make sure to clarify what's going on, thanks for the suggestion.

Reply:
Thanks for pointing this out, turns out the admin check wasn't required, so I've removed it :)